pandas/sparse/tests/test_array.py

```diff
@@ -159,6 +22,58 @@ def setUp(self):
         self.arr = SparseArray(self.arr_data)
         self.zarr = SparseArray([0, 0, 1, 2, 3, 0, 4, 5, 0, 6], fill_value=0)
 
+    def test_constructor_dtype(self):
+        arr = SparseArray([np.nan, 1, 2, np.nan])
+        self.assertEqual(arr.dtype, np.float64)
```

Comment:
no I meant that you are asserting the ``fill_value`` in *some* of the tests here (as well as the dtype), but might want to assert fill_value for all